logsmanagement/flb_plugin.c

```diff
@@ -0,0 +1,1464 @@
+                        syslog_identifier = (char *) p->val.via.str.ptr;
+                        syslog_identifier_size = p->val.via.str.size;
+
+                        new_tmp_text_size += syslog_identifier_size;
+                    }
+                    else if(!strncmp(p->key.via.str.ptr, "PID", (size_t) p->key.via.str.size)){
+                        pid = (char *) p->val.via.str.ptr;
+                        pid_size = p->val.via.str.size;
+
+                        new_tmp_text_size += pid_size;
+                    }
+                    else if(!strncmp(p->key.via.str.ptr, LOG_REC_KEY_SYSTEMD, (size_t) p->key.via.str.size)){
+                        message = (char *) p->val.via.str.ptr;
+                        message_size = p->val.via.str.size;
+
+                        m_assert(message, "message is NULL");
+                        m_assert(message_size, "message_size is 0");
+
+                        new_tmp_text_size += message_size;
+                    }
+                    ++p;
+                    continue;
+                } /* FLB_SYSTEMD or FLB_SYSLOG case end */
+
+                /* FLB_DOCKER_EV case */
+                if(p_file_info->log_type == FLB_DOCKER_EV){ 
+                    if(!strncmp(p->key.via.str.ptr, "time", (size_t) p->key.via.str.size)){
+                        docker_ev_time = p->val.via.i64;
+
+                        m_assert(docker_ev_time, "docker_ev_time is 0");
+                    }
+                    else if(!strncmp(p->key.via.str.ptr, "timeNano", (size_t) p->key.via.str.size)){
+                        docker_ev_timeNano = p->val.via.i64;
+
+                        m_assert(docker_ev_timeNano, "docker_ev_timeNano is 0");
+
+                        if(likely(p_file_info->use_log_timestamp))
+                            timestamp = docker_ev_timeNano / NSEC_PER_MSEC;
+                    }
+                    else if(!strncmp(p->key.via.str.ptr, "Type", (size_t) p->key.via.str.size)){
+                        docker_ev_type = (char *) p->val.via.str.ptr;
+                        docker_ev_type_size = p->val.via.str.size;
+
+                        m_assert(docker_ev_type, "docker_ev_type is NULL");
+                        m_assert(docker_ev_type_size, "docker_ev_type_size is 0");
+
+                        // debug_log("docker_ev_type: %.*s", docker_ev_type_size, docker_ev_type);
+                    }
+                    else if(!strncmp(p->key.via.str.ptr, "Action", (size_t) p->key.via.str.size)){
+                        docker_ev_action = (char *) p->val.via.str.ptr;
+                        docker_ev_action_size = p->val.via.str.size;
+
+                        m_assert(docker_ev_action, "docker_ev_action is NULL");
+                        m_assert(docker_ev_action_size, "docker_ev_action_size is 0");
+
+                        // debug_log("docker_ev_action: %.*s", docker_ev_action_size, docker_ev_action);
```

Comment:
Considering that `debug log` is not used anymore, why didn't you enclose this inside a `macro`? :thinking: They look like good information to debug your code.

Reply:
As discussed, there is a `debug_log()` macro which can be enabled by adding specific flags.